shell/browser/electron_browser_main_parts.cc

```diff
@@ -494,6 +494,13 @@ int ElectronBrowserMainParts::PreMainMessageLoopRun() {
 
 void ElectronBrowserMainParts::WillRunMainMessageLoop(
     std::unique_ptr<base::RunLoop>& run_loop) {
+#if BUILDFLAG(IS_LINUX)
+  auto shutdown_cb = base::BindOnce(run_loop->QuitWhenIdleClosure());
+  ui::OzonePlatform::GetInstance()->PostCreateMainMessageLoop(
+      std::move(shutdown_cb),
```

Comment:
The `shutdown_cb` is called when failed to shutdown, you can just define one like this:

```c++
auto shutdown_cb =
      base::BindOnce([] { LOG(FATAL) << "Failed to shutdown."; });
```

Also chrome put this in `PostCreateMainMessageLoop`, see https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/chrome_browser_main_extra_parts_ozone.cc;l=28